lib/hanami/providers/db/gateway.rb

```diff
@@ -34,6 +35,22 @@ def adapter(name = Undefined)
           end
         end
 
+        # @api public
+        # @since 2.2.0
+        def connection_options(**options)
+          if options.any?
+            config.connection_options.merge!(options)
+          end
+
+          config.connection_options
+        end
+
+        # @api public
+        # @since 2.2.0
+        def options
+          {**connection_options, **adapter.gateway_options}
+        end
```

Comment:
Since these concerns have now been split into two places, I opted to create an `option` method on Gateway to combine them, otherwise I would have to completely duplicate the setting in Adapter and that didn't feel right.